app/src/main/java/org/schabi/newpipe/info_list/dialog/StreamDialogDefaultEntry.java

```diff
@@ -110,6 +116,35 @@ public enum StreamDialogDefaultEntry {
             ShareUtils.shareText(fragment.requireContext(), item.getName(), item.getUrl(),
                     item.getThumbnailUrl())),
 
+    DOWNLOAD(R.string.download, (fragment, item) -> ExtractorHelper.getStreamInfo(
+                    item.getServiceId(),
+                    item.getUrl(),
+                    false
+            )
+            .subscribeOn(Schedulers.io())
+            .subscribe((info) -> {
+                        final ArrayList<VideoStream> sortedVideoStreamsList = new ArrayList<>(
+                                ListHelper.getSortedStreamVideosList(fragment.requireContext(),
+                                        info.getVideoStreams(), null, false, false));
+                        final int index = ListHelper.getDefaultResolutionIndex(
+                                fragment.requireContext(), sortedVideoStreamsList);
+
+                        final DownloadDialog downloadDialog =
+                                DownloadDialog.newInstance(info);
+                        downloadDialog.setVideoStreams(sortedVideoStreamsList);
+                        downloadDialog.setSelectedAudioStream(index);
+                        downloadDialog.setAudioStreams(info.getAudioStreams());
+                        downloadDialog.setSubtitleStreams(info.getSubtitles());
+                        downloadDialog.show(
+                                fragment.getChildFragmentManager(),
+                                "downloadDialog"
+                        );
```

Comment:
Since I saw this [being done in RouterActivity](https://github.com/TeamNewPipe/NewPipe/blob/1e2d76a686b095a65f8052f68e4bbc12625a2108/app/src/main/java/org/schabi/newpipe/RouterActivity.java#L650), are you sure you don't need the following code after calling `show()`? (I sincerely have no idea about it)

```java
fragment.getChildFragmentManager().executePendingTransactions();
```